api/webserver.go

```diff
@@ -0,0 +1,16 @@
+package api
+
+import (
+	"fmt"
+
+	"github.com/gin-gonic/gin"
+)
+
+func RunWebServer(port int) error {
+	addr := fmt.Sprintf(":%d", port)
+	router := gin.Default()
+	router.POST("/user/register", Register)
+	router.Use(AuthMiddleware)
```

Comment:
for reading purposes you need to apply the middleware first

Reply:
but first the user should register to have JWT token and if i place Auth at first place i ran into an error because there is no JWT to be verified?!

Reply:
Yeah You're right